src/functions-and-arrays.js

```diff
@@ -71,3 +157,88 @@ const matrix = [
   [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
   [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
 ];
+
+function greatestProduct(matrix) {
+  if(matrix === null) return null;
+  if(matrix.length === 0) return 0;
+  let product = 1;
+
+  for(let y = 0; y < matrix.length; y++) {
+    for(let x = 0; x < matrix[0].length; x++) {
+      if (getProduct(matrix, x, y) === -1 || getProduct(matrix, x, y) <= product)
+        continue;
+      product = getProduct(matrix, x, y);
+    }
+  }
+  
+  return product;
+}
+
+function getProduct(matrix, x, y) {
+  let productH = 1;
+  let productV = 1;
```

Comment:
The above comment can also be considered here. `productH` and `productV` are not very descriptive just by reading through the code. Consider using one of these conventions to add more description 👉🏾  `horizontalProduct` or `productHorizontal` etc